### PR TITLE
Cleanup loader

### PR DESCRIPTION
**Pull request by privat**
Because #1250 is starting to stall and that my rewrite of the loader is not going well, I started last night an alternative approach: improve the existing loader in small steps.

This PR contains small cleaning on the loader, mainly documentation, small refactorization and bugfixes.
This is a small part of bigger modification still in progress, but I expect these commits to be good enough to be merged.

### CODE REVIEW

File: src/loader.nit [hunk 1]
```diff
@@ -344,16 +344,25 @@ redef class ModelBuilder
 	# See `identify_file`.
 	var identified_files = new Array[ModulePath]
 
-	# Identify a source file
-	# Load the associated project and groups if required
+	# Identify a source file.
+	# Load the associated project and groups if required.
 	#
-	# Silently return `null` if `path` is not a valid module path.
+	# This method does what the used expects when giving an argument to a Nit tool.
```
Comment (R4PaSs) [inline src/loader.nit]: :s/used/user

**Comment by privat** [comment]
reroll: fix typo and a bug (and try to pass the tests)

**Comment by xymus** [comment]
+1

**Comment by privat** [comment]
(try magic jenkins&toaster) test osx please

**Comment by privat** [comment]
re test osx please

**Comment by privat** [comment]
test this please

**Comment by R4PaSs** [comment]
:shipit:

**Pull request merged**